src/deadline/cinema4d_adaptor/Cinema4DAdaptor/adaptor.py

```diff
@@ -392,11 +392,12 @@ def on_run(self, run_data: dict) -> None:
         performs a busy wait until the render completes.
         """
 
+        self.validators.run_data.validate(run_data)
+
         if not self._cinema4d_is_running:
             raise Cinema4DNotRunningError("Cannot render because Cinema4D is not running.")
 
         run_data["frame"] = int(run_data["frame"])
-        self.validators.run_data.validate(run_data)
```

Comment:
Is there a reason we moved this validation up? The line directly above this modifies the run_data, so I would expect it to be validated too?

Reply:
The line above accesses the frame key before validating that it exists. It doesn't really matter in practice - it raises a KeyError or TypeError instead of a ValidationException for malformed data - but logically the data should be validated before it's used.